### PR TITLE
Add our various random implementations to speedtest

### LINKED ISSUE #6492: Ponder the ordering of our random_dns implementations
Out of curiosity I did a silly benchmark (bunch of calls to `dns_random()` in a loop) of the different implementations we have. `getrandom`, `openssl` and `urandom` are in the same ballpark around 80M calls/s but `sodium` is a lot slower, around 1.2 M calls/s. Unfortunately I don't have access to `arc4random` to test it.
I'm not sure this matters for our usage, but we should think about it since `sodium` is currently the default when it's available and `getrandom` is not.

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Fixes #6492 (or at least gives us some data).

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [X] added or modified unit test(s)

On old bullseye laptop:
```
'Random test sodium' 0.38 seconds: 1847879.0 runs/s, 0.54 usec/run
'Random test openssl' 0.12 seconds: 580113.7 runs/s, 1.72 usec/run
'Random test getrandom' 0.34 seconds: 1883117.4 runs/s, 0.53 usec/run
'Random test urandom' 0.51 seconds: 1649074.8 runs/s, 0.61 usec/run
```
Random buster VM:
```
'Random test sodium' 0.48 seconds: 1252499.3 runs/s, 0.80 usec/run
'Random test openssl' 0.12 seconds: 332932.7 runs/s, 3.00 usec/run
'Random test getrandom' 0.54 seconds: 1280666.6 runs/s, 0.78 usec/run
'Random test urandom' 0.45 seconds: 1115434.9 runs/s, 0.90 usec/run
```

on OpenBSD (pretty beefy AMD):
```
'Random test sodium' 0.10 seconds: 35829910.5 runs/s, 0.03 usec/run
'Random test openssl' 0.10 seconds: 30367527.4 runs/s, 0.03 usec/run
'Random test arc4random' 0.10 seconds: 60572491.4 runs/s, 0.02 usec/run
'Random test urandom' 1.26 seconds: 1781943.5 runs/s, 0.56 usec/run
```

All in all I think the current auto order of getrandom, arc4random, libsodium, openssl, urandom is ok.

### CODE REVIEW

**Review by Habbie — APPROVED** [review]

**Comment by Habbie** [comment]
LGTM, but maybe Rémi wants to check why he got different results in #6492.

File: pdns/Makefile.am [hunk 1]
```diff
@@ -916,9 +916,11 @@ tsig_tests_LDADD += $(P11KIT1_LIBS)
 endif
 
 speedtest_SOURCES = \
+        arguments.cc arguments.hh \
```
Comment (Habbie) [inline pdns/Makefile.am:919]: whitespace issue here

File: pdns/Makefile.am [hunk 2]
```diff
@@ -916,9 +916,11 @@ tsig_tests_LDADD += $(P11KIT1_LIBS)
...
 	base32.cc \
 	base64.cc base64.hh \
 	credentials.cc credentials.hh \
+        dns_random.cc dns_random.hh \
```
Comment (Habbie) [inline pdns/Makefile.am:923]: and here

File: pdns/speedtest.cc [hunk 3]
```diff
@@ -1201,10 +1237,17 @@ try
   doRun(StatRingDNSNameQTypeTest(DNSName("example.com"), QType(1)));
 #endif
 
+
```
Comment (Habbie) [inline pdns/speedtest.cc:1240]: and if I'm nitting about whitespace anyway, here too :-)

File: pdns/speedtest.cc [hunk 4]
```diff
@@ -1047,6 +1049,26 @@ struct CredentialsHashTest
...
+    dns_random_init("", true);
+  }
+  string getName() const
+  {
+    return "Random test " + name;
+  }
+
+  void operator()() const
+  {
+    dns_random(0x10000);
```
Comment (rgacogne) [inline pdns/speedtest.cc:1066]:
```suggestion
   dns_random_uint16() ;
```

File: pdns/speedtest.cc [hunk 5]
```diff
@@ -1047,6 +1049,26 @@ struct CredentialsHashTest
...
+  }
+
+  std::string name;
```
Comment (rgacogne) [inline pdns/speedtest.cc:1069]:
```suggestion
  const std::string name;
```

**Review by rgacogne — APPROVED** [review]
Looks good, a few cosmetic suggestions.

As for why I saw such a big difference in #6492, I'd bet my money on df6d5d8cc4293912c5db3f01d245c0b3f78c03d4.

**Pull request merged**